### PR TITLE
SWIFT-481 Support index all paths

### PR DESCRIPTION
**Pull request by ncarbon**
adds the `wildcardProjection` option to `IndexOptions`

### CODE REVIEW

File: Sources/MongoSwift/MongoCollection+Indexes.swift [hunk 1]
```diff
@@ -111,6 +111,9 @@ public struct IndexOptions: Codable {
     /// Optionally specifies fields in the index and their corresponding weight values.
     public var weights: BSONDocument?
 
+    /// Optionally specifies the wildcard projection of a wildcard index.
+    public var wildcardProjection: BSONDocument?
```
Comment (patrickfreed) [inline Sources/MongoSwift/MongoCollection+Indexes.swift:115]: Can we update this to be more more specific? Something like "Optionally specifies a projection document used to determine which fields are indexed by a wildcard index."

Can specify that this feature is only supported in MongoDB 4.2+? 

Also, can we add a link / direct users to dochub.mongodb.org/core/index-wildcard for more information about wild card indexes.

**Review by patrickfreed — COMMENTED** [review]
Code changes LGTM. Just some suggestions for documentation

**Comment by patrickfreed** [comment]
Also, looks like the server changed something such that the wildcardProjects use `true` and `false` instead of 1 and 0.

**Review by patrickfreed — APPROVED** [review]
LGTM pending green ci!

**Review by kmahar — APPROVED** [review]

**Pull request merged**